src/index.js

```diff
@@ -27,14 +27,18 @@ const highlightSearchTerm = ({
   }
   // find all text nodes containing the search term
   const ranges = [];
-  const elements = document.querySelectorAll(selector);
-  Array.from(elements).map((element) => {
-    getTextNodesInElementContainingText(element, search).forEach((node) => {
-      ranges.push(
-        ...getRangesForSearchTermInElement(node.parentElement, search)
-      );
+  try {
+    const elements = document.querySelectorAll(selector);
+    Array.from(elements).map((element) => {
+      getTextNodesInElementContainingText(element, search).forEach((node) => {
+        ranges.push(
+          ...getRangesForSearchTermInElement(node.parentElement, search)
+        );
+      });
     });
-  });
+  } catch (error) {
+    console.error(error);
```

Comment:
please add a comment explaining why it could fail, linking to the difference between `wholeText` and `textContent`